### PR TITLE
API-25817 CCG OAuth Private Key Token Support

### PR DESCRIPTION
**Pull request by dmekala-va**
This PR adds add support for using the private key assertion to actually request a token to the auth-cc.js in addition to the client-id/client-secret functionality that already exists. 

Link to JIRA: https://vajira.max.gov/browse/API-25817

### CODE REVIEW

**Comment by dmekala-va** [comment]
Below shows the request to getting an access token using a private_key is successful:
image

File: auth-cc.js [hunk 1]
```diff
@@ -87,14 +107,39 @@ getMetadata()
     if (!audience) {
       audience = metadata.issuer + "/v1/token";
     }
+    
+    if (argv.clientSecret) {
+      if (!argv.clientId || !argv.clientSecret || !argv.audience) {
```
Comment (dbrown-va) [inline auth-cc.js:112]: With the `argv.clientSecret` confirmed present in the outer `if`, the middle `||` clause can be removed.

File: auth-cc.js [hunk 2]
```diff
@@ -87,14 +107,39 @@ getMetadata()
...
+        argv["client-secret"],
+        audience
+      );
 
-    return getToken(metadata.token_endpoint, assertion);
+      return getToken(metadata.token_endpoint, assertion);
+    }
+
+    if (argv.key) {      
+      if (!argv.clientId || !argv.key || !argv.audience) {
```
Comment (dbrown-va) [inline auth-cc.js:127]: With the `argv.key` confirmed present in the outer `if`, the middle `||` clause can be removed.

**Review by dbrown-va — CHANGES_REQUESTED** [review]
Confirmed functionality, just 2 small changes requested.

**Review by dbrown-va — APPROVED** [review]

**Pull request merged**